src/eof.ts

```diff
@@ -0,0 +1,123 @@
+import { spawn } from "child_process";
+import path from "path";
+import fs from "fs/promises";
+import os from "os";
+import logger from "./logging";
+import { CompileResult, CompilerKind, getCompilerForVersion } from "solc-typed-ast";
+
+// get from ENV or default
+const SOLC_WRAPPER_PATH = process.env.SOLC_WRAPPER_PATH || "solc_wrapper.sh";
+const SOLC_VERSION_WRAPPER_PATH =
+    process.env.SOLC_VERSION_WRAPPER_PATH || "solc_version_wrapper.sh";
+
+const DEFAULT_COMPILER_VERSION = "0.8.27";
+
+let __compilerVersion__: Record<string, string> = {};
+
+export async function compile(sourceCode: string): Promise<CompileResult> {
+    // write source code to temp file
+    const tempFilePath = path.join(os.tmpdir(), "eof_temp.sol");
```

Comment:
Using a fixed filename here could be a problem if we try to parallelize here. Consider using `fs-extra`'s `mkdtempSync()` function here

Reply:
Not a blocker.